scripts/globals/items/amanomurakumo.lua

```diff
@@ -1,13 +1,85 @@
 -----------------------------------------
--- ID: 18318
+-- ID: 18318, 18319, 18646, 18660, 18674, 19755, 19848, 21015, 21016, 21954
 -- Item: Amanomurakumo
 -- Additional Effect: 10% Attack Down
 -----------------------------------------
 require("scripts/globals/status");
 require("scripts/globals/magic");
 require("scripts/globals/msg");
+require("scripts/globals/weaponskills");
+require("scripts/globals/weaponskillids");
 -----------------------------------
 
+-- https://www.bg-wiki.com/bg/Relic_Aftermath
+local aftermathTable = {};
+
+-- Amanomurakumo 75
+aftermathTable[18318] =
+{
+    power=1,
+    duration = function(tp) return math.floor(0.02 * tp); end,
+    mods =
+    {
+        { id=MOD_STORETP, power=7 }
+    }
+};
+aftermathTable[18319] = aftermathTable[18318]; -- Amanomurakumo (80)
+aftermathTable[18646] = aftermathTable[18318]; -- Amanomurakumo (85)
+aftermathTable[18660] = aftermathTable[18318]; -- Amanomurakumo (90)
+aftermathTable[18674] = aftermathTable[18318]; -- Amanomurakumo (95)
+aftermathTable[19755] = aftermathTable[18318]; -- Amanomurakumo (99)
+aftermathTable[19848] = aftermathTable[18318]; -- Amanomurakumo (99/II)
+aftermathTable[21015] = aftermathTable[18318]; -- Amanomurakumo (119)
+aftermathTable[21016] = aftermathTable[18318]; -- Amanomurakumo (119/II)
+
+-- Amanomurakumo (119/III)
+aftermathTable[21954] =
+{
+    power=2,
+    duration = function(tp) return math.floor(0.06 * tp); end,
+    mods =
+    {
+        { id=MOD_STORETP, power=10 },
+        { id=MOD_ZANSHIN, power=10 }
+    }
+};
+
+function onWeaponskill(user, target, wsid, tp, action)
+    if (wsid == WEAPONSKILL_TACHI_KAITEN) then -- Tachi: Kaiten onry
+        local itemId = user:getWeaponID(SLOT_MAIN);
+        if (aftermathTable[itemId]) then
+            -- Apply the effect and add mods
+            addAftermathEffect(user, tp, aftermathTable[itemId]);
+            -- Add a listener for when aftermath wears (to remove mods)
+            user:addListener("EFFECT_LOSE", "AFTERMATH_LOST", aftermathLost);
+        end
+    end
+end
+
+function aftermathLost(target, effect)
+    if (effect:getType() == EFFECT_AFTERMATH) then
+        local itemId = target:getWeaponID(SLOT_MAIN);
+        if (aftermathTable[itemId]) then
+            -- Remove mods
+            removeAftermathEffect(target, aftermathTable[itemId]);
+            -- Remove the effect listener
+            target:removeListener("AFTERMATH_LOST");
+        end
+    end
+end
+
+function onItemCheck(player, param, caster)
+    if (param == ITEMCHECK_EQUIP) then
+        player:addListener("WEAPONSKILL_USE", "AFTERMATH", onWeaponskill);
+    elseif (param == ITEMCHECK_UNEQUIP) then
+        -- Make sure we clean up the effect and mods
+        if (player:hasStatusEffect(EFFECT_AFTERMATH)) then
+            aftermathLost(player, player:getStatusEffect(EFFECT_AFTERMATH))
```

Comment:
i believe this will delete the modifiers twice - once here, and then once again immediately after when the effect is deleted as a result of the TP reset.  Better off just doing nothing here except removing the listener and letting the effect expire on its own

Reply:
How exactly are modifiers deleted on effect wearing? effects/aftermath.lua is going away.

Reply:
Also, I tested it and delModifier was only getting called once when aftermath wore off. Tested both natural wearing and weapon swapping.

Reply:
ah I was a bit confused on the way the listeners were ordered.  I think it would be a tiny bit easier to just delStatusEffect here instead, actually, even though it does get deleted later.  Doing it here should fire AFTERMATH_LOSE while the equip slot is still the weapon, so you wouldn't have to clean up manually here, and then rely on AFTERMATH_LOSE doing nothing when the effect is deleted by the core after

Reply:
Ah okay. That makes sense. I’ll give that a try.

Reply:
So I could have sworn I tested item swapping to make aftermath wear. It's not deleting the mods because the item isn't equipped anymore when it tries to grab the itemId.